src/clients/dotnet/TigerBeetle.Tests/IntegrationTests.cs

```diff
@@ -1130,8 +1132,34 @@ public TBServer()
             format.WaitForExit();
             if (format.ExitCode != 0) throw new InvalidOperationException("format failed");
 
-            process = Process.Start(TB_SERVER, START);
+            process = new Process();
+            process.StartInfo.FileName = TB_SERVER;
+            process.StartInfo.Arguments = START;
+            process.StartInfo.RedirectStandardError = true;
+            process.Start();
             if (process.WaitForExit(100)) throw new InvalidOperationException("Tigerbeetle server failed to start");
+
+            var readPortEvent = new AutoResetEvent(false);
+            var addressLogged = "";
+            var stderrLogged = new StringBuilder();
+            process.ErrorDataReceived += (sender, e) => {
+                var line = e.Data;
+                if (line == null) return;
+                stderrLogged.Append(line);
+                var listening = "listening on ";
+                var found = line.IndexOf(listening);
+                if (found != -1)
+                {
+                    if (addressLogged != "") throw new InvalidOperationException("have already read the port");
+                    addressLogged = line.Substring(found + listening.Length).Trim();
+                    readPortEvent.Set();
+                }
+            };
+            process.BeginErrorReadLine();
+            readPortEvent.WaitOne(60000);

```

Comment:
I think we should call `process.CancelErrorRead()` to stop listening to the stderr after initialization, just in case!

```suggestion
            process.BeginErrorReadLine();
            readPortEvent.WaitOne(60_000);
            process.CancelErrorRead();
```